lib/Index/IndexRecord.cpp

```diff
@@ -513,14 +514,33 @@ emitDataForSwiftSerializedModule(ModuleDecl *module,
   if (*isUptodateOpt)
     return false;
 
+  // Reload resilient modules from swiftinterface to avoid indexing
+  // internal details.
+  bool skipIndexingModule = false;
+  if (module->getResilienceStrategy() == ResilienceStrategy::Resilient) {
+    module->getASTContext().setIgnoreAdjacentModules(true);
+
+    ImportPath::Module::Builder builder(module->getName());
+    auto reloadedModule = module->getASTContext().getModule(builder.get());
+
+    if (reloadedModule) {
+      module = reloadedModule;
+    } else {
+      // If we can't rebuild from the swiftinterface, don't index this module.
+      skipIndexingModule = true;
+    }
+  }
+
   // FIXME: Would be useful for testing if swift had clang's -Rremark system so
   // we could output a remark here that we are going to create index data for
   // a module file.
 
   // Pairs of (recordFile, groupName).
   std::vector<std::pair<std::string, std::string>> records;
 
-  if (!module->isStdlibModule()) {
+  if (skipIndexingModule) {
+    // Don't add anything to records.
```

Comment:
It might be worth adding a comment on the imported modules section below explaining that we believe that's still safe enough to do.

Reply:
I've added that we keep writing the file to avoid rebuilding on broken swiftinterfaces.